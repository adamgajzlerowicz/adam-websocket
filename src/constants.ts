export const getDataMessage = {"event":"subscribe","feed":"book_ui_1","product_ids":["PI_XBTUSD"]}
export const serviceUrl = 'wss://www.cryptofacilities.com/ws/v1'

export const testIds = {
    dataColumnHeading: 'data-column-heading',
    priceHeading: 'price-heading',
    sizeHeading: 'size-heading',
    totalHeading: 'total-heading',
    orderbookPrice: 'orderbook-price',
    orderbookSize: 'orderbook-size',
    orderbookTotal: 'orderbook-total',
    unready: 'unready'
}
