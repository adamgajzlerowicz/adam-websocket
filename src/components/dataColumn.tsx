import * as React from 'react'
import {Col, Row} from "react-grid-system";
import {OrderbookItem} from "./orderbookItem";
import {calculateTotal} from "../utils";
import {PriceSize} from "../types";
import {testIds} from "../constants";

type Props = {
    data: PriceSize
    heading: string
}

export const DataColumn = ({data, heading}: Props) => {
    return <Col xs={12} lg={6}>
        <h3 data-testid={testIds.dataColumnHeading}>{heading}</h3>
        <Row>
            <Col data-testid={testIds.priceHeading}>Price</Col>
            <Col data-testid={testIds.sizeHeading}>Size</Col>
            <Col data-testid={testIds.totalHeading}>Total</Col>
        </Row>
        {data.map((item, index) =>
            <OrderbookItem
                key={index}
                price={item[0]}
                size={item[1]}
                total={calculateTotal(data.slice(index, data.length))}
            />)
        }
    </Col>
}

