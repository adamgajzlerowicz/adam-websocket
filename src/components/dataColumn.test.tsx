import React from "react";
import { render, screen } from "@testing-library/react";
import { DataColumn } from "./dataColumn";
import { testIds } from "../constants";

describe("DataColumn", () => {
  it("renders title", () => {
    render(<DataColumn data={[]} heading="Hello world" />);
    const titleElement = screen.getByTestId(testIds.dataColumnHeading);
    expect(titleElement).toContainHTML("Hello world");
  });

  it("contains headings", () => {
    render(<DataColumn data={[]} heading="Hello world" />);
    expect(screen.getByTestId(testIds.priceHeading)).toBeInTheDocument();
    expect(screen.getByTestId(testIds.sizeHeading)).toBeInTheDocument();
    expect(screen.getByTestId(testIds.totalHeading)).toBeInTheDocument();
  });

  it("renders orderbook correctly", () => {
    const { container } = render(
      <DataColumn
        data={[
          [1, 2],
          [2, 3],
          [3, 4],
        ]}
        heading="Hello world"
      />
    );
    expect(container.innerHTML).toMatchSnapshot();
  });
});
