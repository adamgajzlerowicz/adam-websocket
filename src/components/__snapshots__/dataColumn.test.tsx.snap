// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataColumn renders orderbook correctly 1`] = `"<div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 0px; padding-right: 0px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;\\"><h3 data-testid=\\"data-column-heading\\">Hello world</h3><div style=\\"margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;\\"><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\" data-testid=\\"price-heading\\">Price</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\" data-testid=\\"size-heading\\">Size</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\" data-testid=\\"total-heading\\">Total</div></div><div style=\\"margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;\\"><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">$1.00</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">2.000</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">9.000</div></div><div style=\\"margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;\\"><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">$2.00</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">3.000</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">7.000</div></div><div style=\\"margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;\\"><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">$3.00</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">4.000</div><div style=\\"box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;\\">4.000</div></div></div>"`;
